src/org/rascalmpl/util/maven/SimpleResolver.java

```diff
@@ -90,9 +99,83 @@ private Path calculateJarPath(String groupId, String artifactId, String version,
         return pomLocation.resolveSibling(fileName + ".jar");
     }
 
+    // Note: Because we download both the metadata and the pom files from the first repo that
+    // has the file available, we could end up in a situation where the metadata file comes from 
+    // a different repo than the pom file. This should not pose a problem as pom files with the
+    // same version should be identical. If they are not, there is something seriously wrong with
+    // the versioning in one of the repos and we are helpless to fix that anyway.
+    private Metadata downloadArtifactMetadata(String groupId, String artifactId, String versionSpec)
```

Comment:
normally maven keeps a cache of this meta data in the  `.m2` folder. That seems to be missing? I'm asking this as I don't want us to download this `maven-metadata.xml` file continuously. Especially if for example you don't have a connection, but have all the jars locally.